SRC/seleniumdemotest.py

```diff
@@ -1,48 +1,68 @@
 from selenium import webdriver
 import time
 import pytest
+from selenium.common import NoSuchElementException
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions
 from selenium.webdriver.common.keys import Keys
 import random
+WEBSITE_LINK = "http://seleniumdemo.com/?page_id=7"
+CORRECT_PASSWORD = "#1QAZwsx#1"
+WRONG_PASSWORD = "#1QAZwsx#111"
+CORRECT_LOGIN = ""
 
-user_login = str(random.randint(1, 10000)) + "karikinx@gmail.com"
-password = "#1QAZwsx#1"
-wrong_password = "#1QAZwsx#111"
-def test_register_passed():
+
+def generate_login():
+    return str(random.randint(1, 2000)) + "karikinx@gmail.com"
+
+
+def driver_setup(link):
     driver = webdriver.Chrome()
     driver.maximize_window()
-    driver.get("http://seleniumdemo.com/?page_id=7")
+    driver.get(link)
+    return driver
+
+
+def fill_in_registration_form(driver, login, password):
     driver.find_element(By.ID, "reg_email").click()
-    driver.find_element(By.ID, "reg_email").send_keys(user_login)
+    driver.find_element(By.ID, "reg_email").send_keys(login)
     driver.find_element(By.ID, "reg_password").click()
     driver.find_element(By.ID, "reg_password").send_keys(password)
-    driver.find_element(By.NAME, "register").click()
-    assert driver.find_element(By.LINK_TEXT, "Logout").is_displayed()
-    time.sleep(3)
-
 
-def test_log_in_passed():
-    driver = webdriver.Chrome()
-    driver.maximize_window()
-    driver.get("http://seleniumdemo.com/?page_id=7")
+def fill_in_login_form(driver, login, password):
     driver.find_element(By.ID, "username").click()
-    driver.find_element(By.ID, "username").send_keys(user_login)
+    driver.find_element(By.ID, "username").send_keys(login)
     driver.find_element(By.ID, "password").click()
     driver.find_element(By.ID, "password").send_keys(password)
+
+def test_register_passed():
+    driver = driver_setup(WEBSITE_LINK)
+    user_login = generate_login()
+    fill_in_registration_form(driver, user_login, CORRECT_PASSWORD)
+    driver.find_element(By.NAME, "register").click()
+    time.sleep(1)
+    try:
```

Comment:
i understand that this loop is for creating a user with random prefix untill it will create one that doesnt exist? Is there a better way to ensure that its email is unique ? Or maybe you could think of some solution to rollback database changes after creating a user ? this way there will be no need for all of this

Reply:
this made me wonder can you see the database of this project ? if yes you could write a test to run a query on it to see if the user was created correctly